From 2b36708425ad76f842d8d975713fd6473127caa7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Mon, 18 Nov 2024 10:01:22 +0100
Subject: [PATCH 738/841] fix(stm32mp2): reduce BL2 read only part size on
 trusted boot
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Optimization made to give more space to BL2 RW data in SYSRAM.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 4a07157814905b9680e0031d22ae1c7d967f10e3 fix(stm32mp2): reduce BL2 read only part size on trusted boot]
Change-Id: Idc8473abc935ded803463c11d79aba0e551e31b6
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/419041
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 066caaf71..b2698bc89 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -221,7 +221,7 @@ enum ddr_type {
 #define STM32MP_BL2_RO_SIZE		U(0x00021000)	/* 132 KB */
 #define STM32MP_BL2_SIZE		U(0x0002C000)	/* 176 KB for BL2 */
 #else /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
-#define STM32MP_BL2_RO_SIZE		U(0x00021000)	/* 132 KB */
+#define STM32MP_BL2_RO_SIZE		U(0x00020000)	/* 128 KB */
 #define STM32MP_BL2_SIZE		U(0x0002C000)	/* 176 KB for BL2 */
 #endif /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 #else
-- 
2.47.2

